update pipeline results in transport closing and pachd crash

**What happened?**:
run `pachctl update-pipeline -f <>.json` intermittently results sends message of transport closing .. crashing the pachd pod. 

We have seen this happening 4 times since yesterday. Pachd log is 
```
2018-08-06T06:13:06Z INFO pps.API.call64 {"duration":0.003014112,"request":{"output_commit":{"repo":{"name":"PX"},"id":"2f91fd5dca3841d096657a2b94190b91"},"block_state":true},"response":null} []
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16aae37]

goroutine 17973 [running]:
github.com/pachyderm/pachyderm/src/server/pps/server.(*apiServer).InspectJob(0xc4209e7180, 0x7fd1dfa38208, 0xc420890540, 0xc4209b7b20, 0x0, 0x0, 0x0)
    /go/src/github.com/pachyderm/pachyderm/src/server/pps/server/api_server.go:506 +0x317
github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pps._API_InspectJob_Handler(0x1b94420, 0xc4209e7180, 0x7fd1dfa38208, 0xc420890540, 0xc42047d030, 0x0, 0x0, 0x0, 0xc420dd8ad8, 0xcb08a4)
    /go/src/github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pps/pps.pb.go:3185 +0x241
github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc4200f8840, 0x1dabe80, 0xc420405b00, 0xc4206ac000, 0xc4204ec360, 0x2a417d8, 0x0, 0x0, 0x0)
    /go/src/github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc/server.go:920 +0x848
github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc.(*Server).handleStream(0xc4200f8840, 0x1dabe80, 0xc420405b00, 0xc4206ac000, 0x0)
    /go/src/github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc/server.go:1142 +0x1318
github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4203c6fa0, 0xc4200f8840, 0x1dabe80, 0xc420405b00, 0xc4206ac000)
    /go/src/github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc/server.go:637 +0x9f
created by github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
    /go/src/github.com/pachyderm/pachyderm/src/server/vendor/google.golang.org/grpc/server.go:635 +0xa1
```
**What you expected to happen?**:
update to succeed

**How to reproduce it (as minimally and precisely as possible)?**:
execute update pipeline

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
- Pachyderm CLI version (use `pachctl version`):
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
- OS (e.g. from /etc/os-release):
- Others:

--- a/src/server/pachyderm_test.go
+++ b/src/server/pachyderm_test.go
@@ -8098,6 +8098,30 @@ func TestDatumTries(t *testing.T) {
 	require.Equal(t, tries, observedTries)
 }
 
+func TestInspectJob(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	_, err := pachClient.PpsAPIClient.InspectJob(context.Background(), &pps.InspectJobRequest{})
+	require.YesError(t, err)
+	require.True(t, strings.Contains(err.Error(), "must specify either a Job or an OutputCommit"))
+
+	repo := tu.UniqueString("TestInspectJob")
+	require.NoError(t, c.CreateRepo(repo))
+	_, err = c.PutFile(repo, "master", "file", strings.NewReader("foo"))
+	require.NoError(t, err)
+	ci, err := c.InspectCommit(repo, "master")
+	require.NoError(t, err)
+
+	_, err = c.InspectJobOutputCommit(repo, ci.Commit.ID, false)
+	require.YesError(t, err)
+	require.True(t, strings.Contains(err.Error(), "not found"))
+}
+
 func getObjectCountForRepo(t testing.TB, c *client.APIClient, repo string) int {
 	pipelineInfos, err := pachClient.ListPipeline()
 	require.NoError(t, err)

--- a/src/server/pps/server/api_server.go
+++ b/src/server/pps/server/api_server.go
@@ -494,10 +494,11 @@ func (a *apiServer) InspectJob(ctx context.Context, request *pps.InspectJobReque
 		if request.Job != nil {
 			return nil, fmt.Errorf("can't set both Job and OutputCommit")
 		}
-		if _, err := pachClient.InspectCommit(request.OutputCommit.Repo.Name, request.OutputCommit.ID); err != nil {
+		ci, err := pachClient.InspectCommit(request.OutputCommit.Repo.Name, request.OutputCommit.ID)
+		if err != nil {
 			return nil, err
 		}
-		if err := a.listJob(pachClient, nil, request.OutputCommit, nil, func(ji *pps.JobInfo) error {
+		if err := a.listJob(pachClient, nil, ci.Commit, nil, func(ji *pps.JobInfo) error {
 			if request.Job != nil {
 				return fmt.Errorf("internal error, more than 1 Job has output commit: %v (this is likely a bug)", request.OutputCommit)
 			}

--- a/src/server/pps/server/api_server.go
+++ b/src/server/pps/server/api_server.go
@@ -485,11 +485,17 @@ func (a *apiServer) InspectJob(ctx context.Context, request *pps.InspectJobReque
 	if err := checkLoggedIn(pachClient); err != nil {
 		return nil, err
 	}
+	if request.Job == nil && request.OutputCommit == nil {
+		return nil, fmt.Errorf("must specify either a Job or an OutputCommit")
+	}
 
 	jobs := a.jobs.ReadOnly(ctx)
 	if request.OutputCommit != nil {
 		if request.Job != nil {
 			return nil, fmt.Errorf("can't set both Job and OutputCommit")
+		}
+		if _, err := pachClient.InspectCommit(request.OutputCommit.Repo.Name, request.OutputCommit.ID); err != nil {
+			return nil, err
 		}
 		if err := a.listJob(pachClient, nil, request.OutputCommit, nil, func(ji *pps.JobInfo) error {
 			if request.Job != nil {
@@ -499,6 +505,9 @@ func (a *apiServer) InspectJob(ctx context.Context, request *pps.InspectJobReque
 			return nil
 		}); err != nil {
 			return nil, err
+		}
+		if request.Job == nil {
+			return nil, fmt.Errorf("job with output commit %s not found", request.OutputCommit.ID)
 		}
 	}
 

